src/expectation.jl

```diff
@@ -1,16 +1,16 @@
 const APL = AbstractPolynomialLike
 
-function _expectation(μ::Measure, p::APL, f)
+function _expectation(μ::Measure{T, BT}, p::APL, f) where {T, BT}
     i = 1
     s = 0
-    for t in terms(p)
-        while i <= length(μ.x) && monomial(t) != μ.x[i]
+    for (c, m) in zip(coefficients(p, BT), MB.basis_covering_monomials(BT, monomials(p)))
```

Comment:
`coefficients(p, BT)` computes `maxdegree_basis(Basis, variables(p), maxdegree(p))` which may not match with the basis `basis = MB.basis_covering_monomials(BT, monomials(p))`, use `coefficients(p, basis)` instead maybe

Reply:
See my comment in [MultivariateBases](https://github.com/JuliaAlgebra/MultivariateBases.jl/pull/5#discussion_r439184338).

Reply:
See my comment in [MultivariateBases](https://github.com/JuliaAlgebra/MultivariateBases.jl/pull/5#discussion_r439184338).